### PR TITLE
docs to s3 bucket

### PR DESCRIPTION
**Pull request by mikestonecodes**
##### Checklist

- [x] Linter status: 100% pass
- [x] Changes don't break existing behavior
- [x] Tests coverage hasn't decreased

##### Refers/Fixes

#105 

Automatic doc deployment to S3 Bucket. Uses environment variables  ( can be set with .env file )

AWS_ACCESS_KEY
AWS_SECRET
AWS_REGION
AWS_BUCKET
DOC_TMP_DIR ( if null /tmp/docs-aws )

1) checks out MARKET.js.git 
2) goes through each tag / version and copies 'doc folder' to tmp folder
3) checks out  docs.git  , copies gh-pages branch into same tmp folder
4) uploads to S3 Bucket

triggered in travis 

example:
https://s3.eu-central-1.amazonaws.com/marketjstest/index.html
https://s3.eu-central-1.amazonaws.com/marketjstest/0.2.2/index.html

THIS WILL FAIL RIGHT NOW

needed on deployment server  : environment variables set for correct s3 bucket key etc  ( see above)

tsc installed globally
node installed globally

### CODE REVIEW

**Comment by pelsasser** [comment]
@mikestonecodes - before we merge this, can we remove the logging of the credentials?  Once that is done, I will go ahead and merge and attempt to get travis all squared with the needed env variables to work.

File: tools/update-docs.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+require('dotenv').config();
+echo("Deploying docs!!!");
+echo("Deploying docsaewfjweioafjoew!!!");
+mkdir(tmpfile);
+mkdir(tmpfile + "/alltags");
+cd(tmpfile);
+exec("git init");
+exec("git clone  https://github.com/MARKETProtocol/MARKET.js.git ");
+cd("MARKET.js");
+console.log(process.env.AWS_ACCESS_KEY, process.env.AWS_SECRET, process.env.AWS_REGION);
```
Comment (pelsasser) [inline tools/update-docs.js]: Don't think this is a good idea ;) can we remove!

File: tools/update-docs.ts [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+require('dotenv').config()
+echo("Deploying docs!!!")
+mkdir(tmpfile)
+mkdir(tmpfile+"/alltags")
+cd(tmpfile)
+exec("git init")
+exec(`git clone  https://github.com/MARKETProtocol/MARKET.js.git `);
+cd("MARKET.js")
+
+console.log( process.env.AWS_ACCESS_KEY, process.env.AWS_SECRET, process.env.AWS_REGION)
```
Comment (pelsasser) [inline tools/update-docs.ts]: here too.

**Comment by pelsasser** [comment]
@mikestonecodes - I have env variables set, bucket created, etc.

Node should be good to go, we may need to modify travis for tsc install.

File: tools/update-docs.js (hunk 1 above)
Comment (mikestonecodes) [reply tools/update-docs.js]: LOL whoops.

**Comment by mikestonecodes** [comment]
hey here are the changes I made
1) removed dependency on tsc 
2) removed console logs ( lol sorry )
3) removed retrieving docs repo in favor of this https://github.com/MARKETProtocol/docs/pull/29

**Comment by pelsasser** [comment]
@mikestonecodes looks like this is failing -  https://travis-ci.org/MARKETProtocol/MARKET.js/builds/418842684#L1814

**Pull request merged**